hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/backend/AbstractBackendEntityEnricher.java

```diff
@@ -256,22 +256,39 @@ protected Map<String, org.hypertrace.core.datamodel.AttributeValue> getAttribute
     return attributes;
   }
 
+  // This function was added to enable the backend entity enricher to support updates.
+  // We can also replace this to an update flow if needed.
+  protected Entity mergeBackendEntity(Entity existingEntity, Entity newEntity) {
+    Entity.Builder updatedEntityBuilder = Entity.newBuilder(existingEntity);
+    updatedEntityBuilder.putAllAttributes(newEntity.getAttributesMap());
+    return updatedEntityBuilder.build();
+  }
+
   @Nullable
-  private Entity createBackendIfMissing(Entity backendEntity) {
+  private Entity evaluateBackendEntity(Entity backendEntity) {
     RequestContext requestContext = RequestContext.forTenantId(backendEntity.getTenantId());
     try {
       Optional<Entity> backendFromCache =
           entityCache
               .getBackendIdAttrsToEntityCache()
               .get(requestContext.buildContextualKey(backendEntity.getIdentifyingAttributesMap()));
-      return backendFromCache.orElseGet(
-          () -> {
-            Entity result = this.upsertBackend(backendEntity);
-            LOGGER.info("Created backend:{}", result);
-            return result;
-          });
+      if (backendFromCache.isEmpty()) {
+        Entity result = this.upsertBackend(backendEntity);
+        LOGGER.info("Created backend:{}", result);
+        return result;
+      }
+
+      Entity existingEntity = backendFromCache.get();
+      Entity updatedEntity = mergeBackendEntity(existingEntity, backendEntity);
+      if (!updatedEntity.equals(existingEntity)) {
```

Comment:
Isn't there some last active time kind of an attribute that will always make this check return false and result in an upsert always?

Reply:
There is no time stamp attribute as such. Attribute which can change and can have impact are
```
FROM_EVENT_ID
FROM_EVENT
BACKEND_PATH
http\\u002emethod
```

but this is default implementation only. we are going to override this in other implementations to work with only selected attributes we are concerned about. Should i do the same here for some specific attributes only?

Reply:
Also i can see in L:342
```
final Builder entityBuilder = getEntityBuilder(trace, event, type, backendUri);
```
This first gets the entity and then create the builder. So I don't think these fields will be overridden anytime but sure we cannot assume

Reply:
Yeah. Lets not assume. Lets ensure updates happen in necessary cases.